### PR TITLE
Graph Manipulation Enhancements

### PR DESCRIPTION
**Pull request by chidoneto**
# Flow Expansion and Collapse 

## Default Flow Expansion
Flows in the graph are now recursively open up to a certain *azkaban.depth* value, set in the Azkaban configuration file , with a default value of 2.
flows-expanded-2-levels

# Flow Expansion Commands 
There are 3 new commands in the graph context menu, which allow for the direct recursive flow expansion up to 1, 2 or 3 levels down.
expansionmenu

## Sample Graph with Flow Expansion Context Menu
flow-expansion-menu

## Sample Flow Execution Page with Flow Expansion Context Menu
flow-exec-with-exp-menu

# *Auto Pan Zoom* on Flow Expansion and Collapse
- *Auto Pan Zoom* is now by default preset to ON 
- If *Auto Pan Zoom* is ON, *Reset Pan Zoom* will be automatically applied whenever a flow is collapsed or expanded in the graph.

## Sample graph all collapsed
all-flows-collapsed

## Same sample graph with flow slot-103 expanded and *Auto Pan Zoom* OFF
(Context menu, normally hidden, is shown for demo purposes only)
One-Flow-Exp-APZ-Off

## Sample graph with the same flow expanded and *Auto Pan Zoom* ON
(Context menu, normally hidden, is shown for demo purposes only)
One-Flow-Exp-APZ-ON

### CODE REVIEW

File: azkaban-web-server/src/web/js/azkaban/util/flow-loader.js [hunk 1]
```diff
@@ -188,21 +184,19 @@ var nodeClickCallback = function (event, model, node) {
...
 var jobClickCallback = function (event, model, node) {
-  var target = event.currentTarget;
   var type = node.type;
   var flowId = node.parent.flow;
-  var jobId = node.id;
-
-  var requestURL = contextURL + "/manager?project=" + projectName + "&flow="
-      + flowId + "&job=" + node.id;
-
+  // var jobId = node.id;
```
Comment (ypadron-in) [inline azkaban-web-server/src/web/js/azkaban/util/flow-loader.js:192]: If not used can we delete it?

File: azkaban-web-server/src/web/js/azkaban/util/layout.js [hunk 2]
```diff
@@ -367,7 +367,6 @@ function spaceVertically(layers, maxLayer) {
       }
     }
 
-    console.log("Max " + maxDelta);
```
Comment (ypadron-in) [inline azkaban-web-server/src/web/js/azkaban/util/layout.js:370]: Thank you for deleting this log!

Comment (chidoneto) [reply azkaban-web-server/src/web/js/azkaban/util/layout.js:370]: You bet!

File: azkaban-web-server/src/web/js/azkaban/util/flow-loader.js (hunk 1 above)
Comment (chidoneto) [reply azkaban-web-server/src/web/js/azkaban/util/flow-loader.js:192]: Done!

**Review by ypadron-in — APPROVED** [review]
Thanks for improving user usability!

**Pull request merged**